Attempt at defining initialization, and simulation. (#3146)
modelica/ModelicaSpecification: chapters/equations.tex

\subsection{Equations Needed for Initialization}\label{the-number-of-equations-needed-for-initialization}\label{equations-needed-for-initialization}

\begin{nonnormative}

> In general, for the case of a pure (first order) ordinary differential equation (ODE) system with $n$ state variables and $m$ output variables, we will have $n+m$ unknowns in the simulation problem after the initialization.

Reviewer: We probably shouldn't introduce the term _simulation problem_ here:
```suggestion
In general, for the case of a pure (first order) ordinary differential equation (ODE) system with $n$ state variables and $m$ output variables, we will have $n+m$ unknowns during transient analysis.
```

\end{lstlisting}

Here we have three variables with unknown values: two dynamic variables that also are state variables, \lstinline!x1! and \lstinline!x2!, i.e., $n=2$, one output variable \lstinline!y!, i.e., $m=1$, and one input variable \lstinline!u! with known value.

> A consistent solution of the initialization problem requires initial values for \lstinline!x1!, \lstinline!x2!, \lstinline!der(x1)!, \lstinline!der(x2)!, and \lstinline!y! needs to be found.

Reviewer: Can't have both _requires_ and _needs_:
```suggestion
A consistent solution of the initialization problem requires initial values for \lstinline!x1!, \lstinline!x2!, \lstinline!der(x1)!, \lstinline!der(x2)!, and \lstinline!y! to be found.
```

Author: Good catch, but thinking I believe it is even better to just write 'requires finding initial values for...'.

Reviewer: Yes, here you go:
```suggestion
A consistent solution of the initialization problem requires finding initial values for \lstinline!x1!, \lstinline!x2!, \lstinline!der(x1)!, \lstinline!der(x2)!, and \lstinline!y!.
```

Reviewer: OK, I just found your own suggestion in the comments below.  Resolving.

> A consistent solution of the initialization problem requires initial values for \lstinline!x1!, \lstinline!x2!, \lstinline!der(x1)!, \lstinline!der(x2)!, and \lstinline!y! needs to be found.
> Two additional initial equations thus need to be provided to solve the initialization problem.

Author: ```suggestion
A consistent solution of the initialization problem requires finding initial values for \lstinline!x1!, \lstinline!x2!, \lstinline!der(x1)!, \lstinline!der(x2)!, and \lstinline!y!.
Two additional initial equations thus need to be provided to uniquely specify the initialization problem.
```
'To be found' seemed very passive, and the second sentence was plain wrong; there's no problem solving problems with missing equations - we need the extra equations to uniquely specify the solution.

Reviewer: Maybe we should be careful with saying _uniquely_?  It would be more correct to talk about _well-determined_ or _(globally) balanced_:
```suggestion
A consistent solution of the initialization problem requires finding initial values for \lstinline!x1!, \lstinline!x2!, \lstinline!der(x1)!, \lstinline!der(x2)!, and \lstinline!y!.
Two additional initial equations thus need to be provided to obtain a well-determined (i.e., globally balanced) initialization problem.
```

\end{lstlisting}

Here we have three variables with unknown values: two dynamic variables that also are state variables, \lstinline!x1! and \lstinline!x2!, i.e., $n=2$, one output variable \lstinline!y!, i.e., $m=1$, and one input variable \lstinline!u! with known value.
A consistent solution of the initialization problem requires finding initial values for \lstinline!x1!, \lstinline!x2!, \lstinline!der(x1)!, \lstinline!der(x2)!, and \lstinline!y!.
Two additional initial equations thus need to be provided to obtain a globally balanced initialization problem.

> Additionally those two initial equations must be chosen with care to ensure that they in combination with the dynamic equations give a well-determined initialization problem.

Reviewer: Wouldn't this be even more readable?
```suggestion
Additionally, those two initial equations must be chosen with care to ensure that they give a well-determined initialization problem in combination with the dynamic equations.
```

> Additionally those two initial equations must be chosen with care to ensure that they in combination with the dynamic equations give a well-determined initialization problem.

Author: I prefer the first of those two variants; even if it takes longer to get to the "punch-line", since it more clearly shows that initial and dynamic equations are combined on an (almost) equal footing when making the initialization problem.
